check_shapes/shapes.py

```diff
@@ -78,7 +70,14 @@ def get_shape(shaped: Any, context: ErrorContext) -> Shape:
         if getter is not None:
             return getter(shaped, context)
 
-    raise NoShapeError(StackContext(context, ObjectTypeContext(shaped)))
+    shape = getattr(shaped, "shape", _GET_SHAPE_SENTINEL)
```

Comment:
Potential use case for `for ... else` here:

```suggestion
    for t in inspect.getmro(shaped.__class__):
        getter = _GET_SHAPES.get(t)
        if getter is not None:
            shape = getter(shaped, context)
            break
    else:
        shape = getattr(shaped, "shape", _GET_SHAPE_SENTINEL)
        
    if shape == _GET_SHAPE_SENTINEL:
        raise NoShapeError(StackContext(context, ObjectTypeContext(shaped)))
    assert shape is None or isinstance(shape, tuple), shape
    
    return shape
```